src/renderer.ts

```diff
@@ -152,6 +157,503 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+    const userSort = col.getUserSortBy();
+    const topPadding = Math.max(context.option('rowPadding', 1), 2);
+    const bottomPadding = Math.max(context.option('rowPadding', 1), 2);
+    const dataInfo = col.getDataInfo();
+    const scale = d3.scale.linear().domain([dataInfo.min, dataInfo.max]).range([0, col.getWidth()]);
+
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+      // Rectangle
+      const boxdata = col.getBoxPlotData(d.v, d.dataIndex, scale);
+
+      const sortPosition = {
+        min: boxdata.min,
+        max: boxdata.max,
+        median: boxdata.median,
+        q1: boxdata.q1,
+        q3: boxdata.q3
+      };
+
+      ctx.fillStyle = '#e0e0e0';
+      ctx.strokeStyle = 'black';
+      ctx.beginPath();
+      ctx.rect((boxdata.q1), topPadding, ((boxdata.q3) - (boxdata.q1)), (context.rowHeight(i) - (topPadding + bottomPadding)));
+      ctx.fill();
+      ctx.stroke();
+
+      //Line
+      const minPos = sortPosition.min, maxPos = sortPosition.max, medianPos = sortPosition.median, q3Pos = sortPosition.q3, q1Pos = sortPosition.q1;
+      const bottomPos = (context.rowHeight(i) - bottomPadding);
+      const middlePos = (context.rowHeight(i)) / 2;
+
+      ctx.strokeStyle = 'black';
+      ctx.fillStyle = '#e0e0e0';
+      ctx.beginPath();
+      ctx.moveTo(minPos, middlePos);
+      ctx.lineTo((q1Pos), middlePos);
+      ctx.moveTo(minPos, topPadding);
+      ctx.lineTo(minPos, bottomPos);
+      ctx.moveTo(medianPos, topPadding);
+      ctx.lineTo(medianPos, bottomPos);
+      ctx.moveTo((q3Pos), middlePos);
+      ctx.lineTo(maxPos, middlePos);
+      ctx.moveTo(maxPos, topPadding);
+      ctx.lineTo(maxPos, bottomPos);
+      ctx.stroke();
+      ctx.fill();
+
+
+      if ((col.findMyRanker().getCurrentSortColumn()) === col) {
+
+        ctx.strokeStyle = 'red';
+        ctx.fillStyle = '#ff0700';
+        ctx.beginPath();
+        ctx.moveTo(sortPosition[userSort], topPadding);
+        ctx.lineTo(sortPosition[userSort], bottomPos);
+        ctx.stroke();
+        ctx.fill();
+      } else {
+        return;
+      }
+
+
+    };
+  }
+
+}
+
+class SetCellRenderer implements ICellRendererFactory {
+
+  createSVG(col: SetColumn, context: IDOMRenderContext): ISVGCellRenderer {
+    const cellDimension = col.cellDimension();
+    const binaryValue = col.getConstantValue();
+    return {
+      template: `<g class="upsetcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const circle = d3.select(n).selectAll('circle').data(<any>col.getValue(d.v, d.dataIndex));
+        circle.enter().append('circle');
+        circle
+          .attr({
+            cy: (d: any, i) => (context.rowHeight(i) / 2),
+            cx: (d: any, i) => (i * cellDimension) + (cellDimension / 2),
+            r: (cellDimension / 4),
+            class: 'upsetcircle',
+            opacity: (d) =>(d === binaryValue) ? 1 : 0.1
```

Comment:
I would use a different css class for the different types to be better styleable